beacon-chain/sync/rpc_send_request.go

```diff
@@ -129,3 +134,65 @@ func SendBeaconBlocksByRootRequest(
 	}
 	return blocks, nil
 }
+
+func SendBlobSidecarByRoot(
+	ctx context.Context, tor blockchain.TemporalOracle, p2pApi p2p.P2P, pid peer.ID,
+	ctxMap ContextByteVersions, req *p2ptypes.BlobSidecarsByRootReq,
+) ([]*pb.BlobSidecar, error) {
+	topic, err := p2p.TopicFromMessage(p2p.BlobSidecarsByRootName, slots.ToEpoch(tor.CurrentSlot()))
+	if err != nil {
+		return nil, err
+	}
+	log.WithField("topic", topic).Debug("Sending blob sidecar request")
+	stream, err := p2pApi.Send(ctx, req, topic, pid)
+	if err != nil {
+		return nil, err
+	}
+	defer closeStream(stream, log)
+
+	return readChunkEncodedBlobs(stream, p2pApi.Encoding(), ctxMap)
+}
+
+var ErrBlobChunkedReadFailure = errors.New("failed to read stream of chunk-encoded blobs")
+var ErrBlobUnmarshal = errors.New("Could not unmarshal chunk-encoded blob")
+
+func readChunkEncodedBlobs(stream network.Stream, encoding encoder.NetworkEncoding, ctxMap ContextByteVersions) ([]*pb.BlobSidecar, error) {
+	decode := encoding.DecodeWithMaxLength
+	max := int(params.BeaconNetworkConfig().MaxRequestBlobsSidecars)
+	var (
+		code uint8
+		msg  string
+		err  error
+	)
+	sidecars := make([]*pb.BlobSidecar, 0)
+	for i := 0; i < max; i++ {
+		code, msg, err = ReadStatusCode(stream, encoding)
+		if err != nil {
+			break
+		}
+		if code != 0 {
+			return nil, errors.Wrap(ErrBlobChunkedReadFailure, msg)
+		}
+		ctxb, err := readContextFromStream(stream)
+		if err != nil {
+			return nil, errors.Wrap(err, "error reading chunk context bytes from stream")
+		}
+
+		v, found := ctxMap[bytesutil.ToBytes4(ctxb)]
+		if !found {
+			return nil, errors.Wrapf(ErrBlobUnmarshal, fmt.Sprintf("unrecognized fork digest %#x", ctxb))
+		}
+		if v != version.Deneb {
+			return nil, fmt.Errorf("unexpected context bytes for deneb BlobSidecar, ctx=%#x, v=%s", ctxb, version.String(v))
+		}
+		sc := &pb.BlobSidecar{}
+		if err := decode(stream, sc); err != nil {
+			return nil, errors.Wrap(err, "failed to decode the protobuf-encoded BlobSidecar message from RPC chunk stream")
+		}
+		sidecars = append(sidecars, sc)
```

Comment:
there seems to be no validation on the blob sidecars, how do we verify that the blobs being provided are the ones we want ?

Reply:
The response may not include all requested values:
```
Clients MUST respond with at least one sidecar, if they have it. Clients MAY limit the number of blocks and sidecars in the response.
```
Since the caller of this function may need to formulate a new request to get the missing values, I thought it was better to leave that kind of logic to the user of the function. What do you think?

Reply:
I think that is fine, but is there any correctness check that we can do to verify that the sidecars provided by the peer are at least a subset of what we requested ? Thinking about how to protect us in the event of malicious peer just returning us nonsense